### PR TITLE
New OP_TYPE_IS_COP_NN() macro

### PR DESCRIPTION
**Pull request by leonerd**
I'm doing a lot of hackery in peep.c and op.c, and on occasions need to detect if an op is OP_NEXTSTATE (or OP_DBSTATE). Basically, almost any time you care about OP_NEXTSTATE you need to also check for OP_DBSTATE as well, so it's a lot easier to avoid mistakes if you just have a macro that checks for either.

There's some leftover code in a few places that still checks for `->op_type == OP_NEXTSTATE` directly without also caring about OP_DBSTATE and honestly I don't know if those are bugs. But with this macro in place, it's easier to find those now.

* This set of changes does not require a perldelta entry.

### CODE REVIEW

File: peep.c [hunk 1]
```diff
@@ -3379,8 +3376,7 @@ Perl_rpeep(pTHX_ OP *o)
                             && (p->op_private & OPpLVAL_INTRO) == intro
                             && !(p->op_private & ~OPpLVAL_INTRO)
                             && p->op_next
-                            && (   p->op_next->op_type == OP_NEXTSTATE
-                                || p->op_next->op_type == OP_DBSTATE)
```
Comment (tonycoz) [inline peep.c:3383]: Changed from checking `p->op_next` to `p`.

Comment (leonerd) [reply peep.c:3383]: Oops, so it is, good catch. Will force-push

**Comment by tonycoz** [comment]
`gv.c`, `pp_hot.c` and `regexec.c` also do this check.

In all of these places you've added a NULL check which wasn't there before, ie. from:
```
op->op_type == OP_NEXTSTATE || op->op_type == OP_DBSTATE
```
to:
```
(op && op->op_type == OP_NEXTSTATE) || (op && op->op_type == OP_DBSTATE)
```
which I'm sure Coverity will complain about (the check means op might be null, but we use it without checking in other places)

File: peep.c (hunk 1 above)
Comment (tonycoz) [reply peep.c:3383]: Failing tests caught it before I noticed it :)

**Comment by leonerd** [comment]
Ohyeah, Hrmmm.. That's a subtle part of `OP_TYPE_IS` that I hadn't realised before. I'll make them all `_NN` instead for now.

**Comment by leonerd** [comment]
> `gv.c`, `pp_hot.c` and `regexec.c` also do this check.

I found one in gv.c, one in regexec.c. None in pp_hot.c, but I did find a few in pp_ctl.c.

**Comment by leonerd** [comment]
There's still a few other tests that might be tidied up by a new `OP_TYPE_IS_OR_WAS_COP()` but maybe I'll add that in a separate commit later.

**Comment by tonycoz** [comment]
> None in pp_hot.c, but I did find a few in pp_ctl.c.

Sorry, yes.

**Review by tonycoz — APPROVED** [review]

**Pull request merged**